File: java/leetcode_grind/Day721.java
package leetcode_grind;

public class Day721 {
    // https://leetcode.com/problems/largest-combination-with-bitwise-and-greater-than-zero/description/?envType=daily-question&envId=2024-11-07
    static class Solution1 {
        public int largestCombination(int[] candidates) {
            int[] bitCount = new int[24];
            for (int i = 0; i < 24; i++) {
                for (int num : candidates) {
                    if ((num & (1 << i)) != 0) {
                        bitCount[i]++;
                    }
                }
            }

            int max = 0;
            for (int count : bitCount) {
                if (count > max) {
                    max = count;
                }
            }
            return max;
        }
    }

}

File: src/leetcode_grind/day_721.rs
// https://leetcode.com/problems/largest-combination-with-bitwise-and-greater-than-zero/description/?envType=daily-question&envId=2024-11-07
pub fn largest_combination(candidates: Vec<i32>) -> i32 {
    let mut max_count = 0;
    for i in 0..24 {
        let mut count = 0;
        for &num in &candidates {
            if (num & (1 << i)) != 0 {
                count += 1;
            }
        }
        max_count = max_count.max(count);
    }
    max_count
}
